temporian/implementation/numpy/operators/arithmetic/floordiv.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from temporian.implementation.numpy.operators.arithmetic.base import (
+    BaseArithmeticNumpyImplementation,
+)
+from temporian.core.operators.arithmetic import FloorDivideOperator
+from temporian.implementation.numpy import implementation_lib
+
+
+class FloorDivideNumpyImplementation(BaseArithmeticNumpyImplementation):
+    """Actual numpy implementation to divide of first event by the second one
+    and take the result floor"""
+
+    def __init__(self, operator: FloorDivideOperator) -> None:
+        super().__init__(operator)
+
+    def _do_operation(self, event_1_feature, event_2_feature):
+        return event_1_feature.data // event_2_feature.data
```

Comment:
You ended the normal division with this to ensure same types:

```python
.astype(
            event_1_feature.data.dtype
        )
```
Why is it not here? Is this needed on the / and not on //? or is not needed on both?

Reply:
Nice catch. It was needed only in `/` because numpy would promote ints to floats automatically. Now, since it's forbidden the integer division, it's not needed there either (I'll remove it).
With this `//` operator, the output type is always the same as the input type.

Reply:
Float // float does still return float right?